error.go

```diff
@@ -23,11 +25,13 @@ import (
 func IsErrorWithCode(err error, code int) bool {
 
 	var c int
-	switch e := err.(type) {
-	case Error:
-		c = e.Code
-	case Errors:
-		c = e.Code()
+	var e1 Error
+	var e2 Errors
+
+	if errors.As(err, &e1) {
+		c = e1.Code
+	} else if errors.As(err, &e2) {
+		c = e2.Code()
```

Comment:
great! this has irked me for a long time. So much that I nearly made a PR for this on Monday! :D